### DISCUSSION TITLE
【四月新功能汇总】+五月上新前瞻！

### DISCUSSION THREAD

**Discussion by cocorunning** [Tencent/CodeAnalysis#263] [Announcements]
![四月上线（黑）](https://user-images.githubusercontent.com/91463108/165241690-521e1e28-9daf-47ca-a797-6149a34a2c1c.png)

### 【上新一：上新合规检查规则包】

此规则包在内部使用多年，现已在开源版上线。帮助简化开源合规流程，助力规避版权、安全、公关等问题。
[[详情可查看专题文章：开源合规检查]](https://github.com/Tencent/CodeAnalysis/discussions/240)

### 【上新二：新增健康状态勘测功能】

增加健康探测接口，在启动前后探测各项内容是否正常启动。
​在启动前对python命令及版本、所需依赖包是否安装、pip下载源、可用磁盘空间、可用内存等进行探测；启动后通过访问健康状况探测接口监测服务健康状况及异步任务执行状态。

### 【上新三：新增部分PHP安全相关规则】

有限支持：SQL注入, XSS, Deser反序列化, SSRF, 会话固定, PHPINFO, URL重定向, FileInclusion等安全漏洞检查。

### 【上新四：上线工具管理】

1.支持接入自定义工具。

（工具权限默认为仅团队内可用，如需调整工具权限需平台管理员处理）

![支持接入自定义工具](https://user-images.githubusercontent.com/91463108/165242626-e8f18b54-cb34-456a-add0-9a0634e8465c.png)

2.支持为自定义工具创建规则。

![创建规则](https://user-images.githubusercontent.com/91463108/165242713-da9e595d-d7c1-4646-905c-f635ecfd5b69.png)

3.支持编辑自定义工具基础信息及运营状态。

![基础信息和运营状态](https://user-images.githubusercontent.com/91463108/165242733-46a46fa0-5338-4b4a-a1c2-8634a47491d0.png)

### 【上新五：License鉴权】

支持部分子工具的License鉴权。

### 【上新六：增加工具管理（管理员权限）】

1.支持查看平台已支持的工具列表。

![工具列表](https://user-images.githubusercontent.com/91463108/165242960-425465ae-43d2-4569-8325-815c705b10ea.png)

2.支持调整自定义工具权限。

<img width="1782" alt="自定义工作权限" src="https://user-images.githubusercontent.com/91463108/165242990-e9a7219e-3635-4aa2-aa1f-1876136a24d3.png">

以上功能代码已提交开源版，欢迎使用！下月更新功能前瞻，敬请期待！👇👇

### 【五月上新预告】（各项功能和内容以上线后版本介绍为准）

1.上线工具依赖管理
可自定义工具依赖项，拉取工具依赖时仅拉取指定项，提高任务效率。

2.支持自定义分析节点私有标签
可自定义标签实现项目私有机器分类管理。

3.支持各主流SCM平台Oauth授权，便捷高效接入主流SCM平台代码库。

4.自研工具正式版开放使用
支持更多语言，支持自定义规则。

5.支持API扫描，助力隐私合规分析。

**欢迎使用！**

**Comment by cyw3** [comment]
NICE!

**Comment by owlmk** [comment]
LGTM
